### ISSUE TITLE
`manual_clamp` improvement for unsigned types

### ISSUE THREAD

**Issue by LesnyRumcajs** [rust-lang/rust-clippy#16792]
### Description

I noticed the `manual_clamp` doesn't trigger on some potential lint opportunity:
```rust
fn unsigned_clamp(mut x: u32) -> u32 {
    if x == 0 {
        x = 1;
    } else if x > 5 {
        x = 5
    }
    x
}
```

This passes the check, but for unsigned types we could do some normalization that would effectively turn it to:
```rust
fn unsigned_clamp(mut x: u32) -> u32 {
    if x < 1 {
        x = 1;
    } else if x > 5 {
        x = 5
    }
    x
}
```
and trigger the warning.

Before jumping into implementation I'd like to get some feedback if it's something actually useful to add; it's a bit of a specific case but I _guess_ it's not that seldom that one wants to avoid zero in their unsigned calculations.

This could be further generalized to `T::MIN` and `T::MAX` normalization in case of adjacent bounds, but might be an overkill (though not necessarily more involved to implement)

Would this be useful?

### Version

```text
rustc 1.94.1 (e408947bf 2026-03-25)
binary: rustc
commit-hash: e408947bfd200af42db322daf0fadfe7e26d3bd1
commit-date: 2026-03-25
host: x86_64-unknown-linux-gnu
release: 1.94.1
LLVM version: 21.1.8
```

### Additional Labels

@rustbot C-question

**Comment by J-ZhengLi** [comment]
> it's a bit of a specific case

yes, it could be considered as an edge case for unsigned integers, so I'd say it's a valid idea.

Also, you are very welcomed to ask this kind of questions in our [zulip channel](https://rust-lang.zulipchat.com/#narrow/channel/257328-clippy) 😄 

edit: 
> This could be further generalized to T::MIN and T::MAX normalization...

forgive me for multiple edits, cuz I was a little confused about this, if I understand correctly, you want to take the evaluated constant value out of `MIN` or `MAX` then see if a value of `min + 1` or `max - 1` was assigned? If yes, well it is indeed an overkill because such assumption only works for uint (I think).

**Comment by LesnyRumcajs** [comment]
> forgive me for multiple edits, cuz I was a little confused about this, if I understand correctly, you want to take the evaluated constant value out of MIN or MAX then see if a value of min + 1 or max - 1 was assigned? If yes, well it is indeed an overkill because such assumption only works for uint (I think).

I think it's fine for signed as well, but I'm having a hard time imagining someone doing this; they'd need to be clamping, e.g., from `i32::MIN + 1`.

> Also, you are very welcomed to ask this kind of questions in our [zulip channel](https://rust-lang.zulipchat.com/#narrow/channel/257328-clippy) 😄

I had a feeling you had a channel somewhere, I recalled it from some conference talk, but couldn't quickly find it. Would it make sense to put it in README? edit: I see it now in the `CONTRIBUTING.md`. I guess I should have read that in the first place!
